In [24]:
import pandas as pd
import numpy as np


In [32]:
inscritos_ENEM = ['NU_INSCRICAO']
df_inscritos = pd.read_csv('datasets/MICRODADOS_ENEM_2020.csv', sep=";", encoding="ISO-8859-1", usecols=inscritos_ENEM)
df_inscritos.isna().any()

NU_INSCRICAO    False
dtype: bool

In [19]:
df_inscritos.shape

(5783109, 1)

In [ ]:
#enem_2020 = ['TP_LINGUA', 'TP_FAIXA_ETARIA', 'TP_SEXO', 'NO_MUNICIPIO_ESC', 'SG_UF_ESC', 'TP_DEPENDENCIA_ADM_ESC', 'TP_LOCALIZACAO_ESC', 'NO_MUNICIPIO_PROVA', 'SG_UF_PROVA']
#data_ENEM = pd.read_csv('datasets/MICRODADOS_ENEM_2020.csv', sep=";", encoding="ISO-8859-1", usecols=enem_2020)

In [98]:
notas_ENEM = ['NU_NOTA_CN', 'NU_NOTA_CH', 'NU_NOTA_LC', 'NU_NOTA_MT', 'NU_NOTA_REDACAO']
df_notas = pd.read_csv('datasets/MICRODADOS_ENEM_2020.csv', sep=";", encoding="ISO-8859-1", usecols=notas_ENEM)

In [99]:
df_notas.isna().any()

NU_NOTA_CN         True
NU_NOTA_CH         True
NU_NOTA_LC         True
NU_NOTA_MT         True
NU_NOTA_REDACAO    True
dtype: bool

In [105]:
(df_notas.isnull().sum() / df_notas.shape[0])*100

NU_NOTA_CN         55.085751
NU_NOTA_CH         52.376135
NU_NOTA_LC         52.376135
NU_NOTA_MT         55.085751
NU_NOTA_REDACAO    52.376135
dtype: float64

In [104]:
df_notas.shape[0]

5783109

In [36]:
df_notas.dropna(axis=0, subset=['NU_NOTA_CN', 'NU_NOTA_CH', 'NU_NOTA_LC', 'NU_NOTA_MT', 'NU_NOTA_REDACAO'], inplace=True)

In [48]:
df_notas['NU_NOTA_MT'].mean()

520.7272992307676

In [115]:
escolas = ['SG_UF', 'NO_MUNICIPIO', 'NO_ENTIDADE', 'TP_DEPENDENCIA', 'TP_LOCALIZACAO']
df_escolas = pd.read_csv('datasets/microdados_ed_basica_2021.csv', sep=";", encoding="latin-1", usecols=escolas)

In [116]:
df_escolas_selecionadas = df_escolas.loc[(df_escolas['SG_UF'] == 'MA') & (df_escolas['TP_LOCALIZACAO'] == 1) & (df_escolas['TP_DEPENDENCIA'] != 4), escolas]

In [117]:
df_EE_selecionadas = df_escolas.loc[(df_escolas['SG_UF'] == 'MA') & (df_escolas['TP_LOCALIZACAO'] == 1) & (df_escolas['TP_DEPENDENCIA'] == 2), escolas]

In [118]:
df_EE_Rural_selecionadas = df_escolas.loc[(df_escolas['SG_UF'] == 'MA') & (df_escolas['TP_LOCALIZACAO'] == 2) & (df_escolas['TP_DEPENDENCIA'] == 2), escolas]

In [119]:
df_EM_selecionadas = df_escolas.loc[(df_escolas['SG_UF'] == 'MA') & (df_escolas['TP_LOCALIZACAO'] == 1) & (df_escolas['TP_DEPENDENCIA'] == 3), escolas]

In [120]:
df_EM_Rural_selecionadas = df_escolas.loc[(df_escolas['SG_UF'] == 'MA') & (df_escolas['TP_LOCALIZACAO'] == 2) & (df_escolas['TP_DEPENDENCIA'] == 3), escolas]

In [123]:
df_EM_Rural_selecionadas

SG_UF NO_MUNICIPIO                    NO_ENTIDADE  TP_DEPENDENCIA  \
26101    MA   Açailândia        EM MARIO CABRAL DE MELO               3   
26104    MA   Açailândia       EM ANTONIO ALVES MOREIRA               3   
26105    MA   Açailândia            EM ANTONIO FERREIRA               3   
26108    MA   Açailândia          EM DIVINA PROVIDENCIA               3   
26111    MA   Açailândia  EM JUSTINO GUSMAO DE OLIVEIRA               3   
...     ...          ...                            ...             ...   
40233    MA      Zé Doca                     EM PIO XII               3   
40235    MA      Zé Doca            EM SAO JOAO BATISTA               3   
40236    MA      Zé Doca             EM PRINCESA CALDAS               3   
40238    MA      Zé Doca               EM NOSSA VITORIA               3   
40239    MA      Zé Doca                 EM SANTA CLARA               3   

       TP_LOCALIZACAO  
26101               2  
26104               2  
26105               2  
26108               2  
26111               2  
...               ...  
40233               2  
40235               2  
40236               2  
40238               2  
40239               2  

[8638 rows x 5 columns]

In [91]:
df_EE_selecionadas.to_excel('Escolas_Estaduais.xlsx', index=False)
df_EE_Rural_selecionadas.to_excel('Escolas_Estaduais_Rurais.xlsx', index=False)
df_EM_selecionadas.to_excel('Escolas_Municipais.xlsx', index=False)
df_EM_Rural_selecionadas.to_excel('Escolas_Municipais_Rurais.xlsx', index=False)

# Missing

In [97]:
df_escolas.isnull().sum()

SG_UF             0
NO_MUNICIPIO      0
NO_ENTIDADE       0
TP_DEPENDENCIA    0
TP_LOCALIZACAO    0
dtype: int64

IDHM_dataset.xlsx
0%

microdados_ed_basica_2021.csv
0%

MICRODADOS_ENEM_2020.csv
55.09% para os dados utilizados (notas)